### ISSUE TITLE
Invalid override in subclass accepted

### ISSUE THREAD

**Issue by clo-vis** [python/mypy#19066]
**Bug Report**

**To Reproduce**

```python
from dataclasses import dataclass


@dataclass
class P:
    name: str | None


@dataclass
class Q(P):
    name: str


class A:
    def __init__(self) -> None:
        self._name: str | None = None

    @property
    def name(self) -> str | None:
        return self._name

    @name.setter
    def name(self, v: str | None) -> None:
        self._name = v

class B(A):
    def __init__(self) -> None:
        self._name = ""

    @property
    def name(self) -> str:
        assert self._name is not None
        return self._name

    @name.setter
    def name(self, v: str) -> None:
        self._name = v


def f(obj: P | A) -> None:
    obj.name = None


q = Q(name="")
b = B()
f(q)
f(b)
```

**Expected Behavior**

mypy should warn me that overriding name (in B and Q) is invalid.

(pyright for example says: 
```
"name" overrides symbol of same name in class "P"
  Variable is mutable so its type is invariant
    Override type "str" is not the same as base type "str | None"
```
)

**Actual Behavior**

No warning.

**Your Environment**

- Mypy version used: 1.15.0
- Mypy command-line flags: 
- Mypy configuration options from `mypy.ini` (and other config files): strict = true
- Python version used: 3.12

**Comment by sterliakov** [comment]
By default `mypy` allows covariant overrides of mutable attributes. That's unsafe, but too common to prohibit. You can enable an extra error code to report these violations (`--enable-error-code=mutable-override`). It's documented [here](https://mypy.readthedocs.io/en/stable/error_code_list2.html#check-that-overrides-of-mutable-attributes-are-safe-mutable-override), here's a [playground](https://mypy-play.net/?mypy=1.15.0&python=3.12&flags=strict&enable-error-code=mutable-override&gist=db22422d57b267d586bf92a0fd5929a4) to demonstrate.
